Modules/galaxy/galaxy.py

```diff
@@ -0,0 +1,256 @@
+"""
+galaxy.py - Provide functions to interact with and retrieve useful information from the
+            Fuel Rats Systems API.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+
+from html import escape
+import json
+from typing import Dict, List, Optional
+
+import aiohttp
+
+from config import config
+from .star_system import StarSystem
+
+
+class Galaxy:
+    """
+    Worker class to interface with the Fuel Rats Systems API.
+    """
+
+    MAX_PLOT_DISTANCE = 20000
+    SCOOPABLE_SPECTRAL_CLASSES = ('K', 'G', 'B', 'F', 'O', 'A', 'M')
+
+    def __init__(self, url: str = None):
+        self.url = url or config['api']['url']
+
+    async def find_system_by_name(self, name: str) -> Optional[StarSystem]:
+        """
+        Finds a single system by its name and return its StarSystem object
+
+        Args:
+            name (str): The name of the system to search for.
+
+        Returns:
+            A ``StarSystem`` object representing the found system, or ``None`` if none was found.
+        """
+
+        data = await self._call("systems", {"filter[name:like]": name.upper(), "include": "bodies"})
+        result_count = data['meta']['results']['available']
+        if result_count > 0:
+            sys = data['data'][0]['attributes']
+            star_name = f"{sys['name']} A".casefold()
+            bodies = data['included']
+            for body in bodies:
+                body_name = body['attributes']['name'].casefold()
+                if (body_name == sys['name'].casefold() or body_name == star_name):
+                    sys['spectral_class'] = body['attributes']['spectral_class']
+            return StarSystem.from_dict(sys)
```

Comment:
yeah... i need more comments in this, Im having a hard time understanding the logic here.

Reply:
if the body name is the same as the system name or the known star name, this body is the main star, so the systems spectral class is set to the spectral class of the body

Reply:
not that kind of comment. I mean documentation in the code to make it less dense and more comprehensible.

Reply:
that code is comprehensible. It didn't even set off the CC warning for too high CC!!!!11elf /s
But yeah, it could use one or two comments explaining the assumptions.